static/js/stream_edit.js

```diff
@@ -676,6 +676,14 @@ exports.initialize = function () {
         }
 
         subs.sub_or_unsub(sub);
+
+        // If the user click on the Subscribe button in
+        // nonsubbed_stream_narrow_message, change the banner
+        // to empty_narrow_message from nonsubbed_stream_narrow_message.
+        if ($(".stream_sub_unsub_button").closest("#nonsubbed_stream_narrow_message").length) {
+            $("#empty_narrow_message").show();
+            $("#nonsubbed_stream_narrow_message").hide();
+        }
```

Comment:
Thanks for working on this @nikhilmaske-2001!

I think this change isn't correct, in that if the user subscribed instead via the "Manage streams" modal (or another browser window), the result is we'd end up with the wrong state documented in the issue as well.

I think we need to change the `stream_events.js` handler to trigger this update, e.g. "Find any `sub_unsub_button` s for the modified stream ID and toggle their state".

Reply:
@timabbott I have pushed the changes, please verify whether this fix is correct or not.